cli/tests/activate/nested-activate-fish.exp

```diff
@@ -0,0 +1,40 @@
+# Activate a project environment using --dir
+
+set dir [lindex $argv 0]
+set flox $env(FLOX_BIN)
+set timeout 10
+
+spawn $flox activate --dir $dir/outer
+expect_after {
+  timeout { exit 1 }
+  eof { exit 2 }
+  "*\n" { exp_continue }
+  "*\r" { exp_continue }
+}
+expect "You are now using the environment"
+
+send "$flox activate --dir $dir/inner\r"
+expect "You are now using the environment"
+
+send "inner\r"
+expect "the inner alias is defined"
+
+send "outer\r"
+expect "the outer alias is defined"
+
+send "current\r"
+expect "the current alias is inner"
+
+puts "INFO: exiting inner env"
+send "exit 0\r"
+
+puts "INFO: exiting outer env"
+send "exit 0\r"
+
+# Would normally wait for eof but does not work with fish?
+# puts "INFO: waiting for eof"
+# expect eof
+# puts "TEST COMPLETE: received eof"
```

Comment:
suggestion followup: to me that smells like something is going wrong and we should figure out what

Reply:
Spoke with Matthew about this and the best theory is that fish is remiss in not relaying EOF to the calling process as it exits (which in this case is `expect`). It's strange that it only happens on Linux in CI, but at the same time it's not a big return on the time investment required to investigate further so we've corrected for the behavior by adding a parameter to control for this expectation in the `nested-activate-fish.exp` script.